cli/src/parse.rs

```diff
@@ -100,24 +100,43 @@ pub fn parse_file_at_path(parser: &mut Parser, opts: &ParseFileOptions) -> Resul
     let time = Instant::now();
 
     #[inline(always)]
-    fn is_utf16_bom(bom_bytes: &[u8]) -> bool {
-        bom_bytes == [0xFF, 0xFE] || bom_bytes == [0xFE, 0xFF]
+    fn is_utf16_le_bom(bom_bytes: &[u8]) -> bool {
+        bom_bytes == [0xFF, 0xFE]
     }
 
+    #[inline(always)]
+    fn is_utf16_be_bom(bom_bytes: &[u8]) -> bool {
+        bom_bytes == [0xFE, 0xFF]
+    }
+
+    // if encoding is none and it's got an LE or BE bom, assign to LE or BE
+    opts.encoding = match opts.encoding {
```

Comment:
Nitpick, but would it be better to assign the chosen encoding to a local here and pass `opts` as an immutable reference? Doesn't seem to make a difference in performance (tried with some large and small files) but imo does help with readability (since that's the only part of the struct being modified).


<details>
<summary>Bench</summary>

![image](https://github.com/user-attachments/assets/8d135b73-fe53-44ad-a483-c7c62e41aca5)

</details>